Enhance 'purge' command with additional options

dotpmm opened the pull request:
Updated the 'purge' command to allow deletion by amount, date, or message link.

## 📌 Description
Updated the purge command to support additional deletion methods based on moderator needs.
This PR implements three mutually exclusive purge options:

- Amount – delete the last N messages
- Message Link – delete all messages newer than a specific message
- Date – delete all messages from a given date (DD-MM-YYYY) forward

This enhancement resolves: https://github.com/pesu-dev/discord_bot/issues/65
---

## 🧱 Type of Change

- [ ] 🐛 Bug fix – Non-breaking fix for a functional/logic error
- [x] ✨ New feature – Adds functionality without breaking existing commands or APIs
- [ ] ⚠️ Breaking change – Backward-incompatible change (commands, bot behavior, etc.)
- [ ] 📝 Documentation update – README, comments, help text, etc.
- [ ] 🧪 Test suite change – Adds/updates unit, integration, or manual tests
- [ ] ⚙️ CI/CD pipeline update – GitHub Actions, Docker, Git hooks, etc.
- [ ] 🧹 Refactor – Code cleanup, improvements, or style changes
- [ ] 🐢 Performance improvement – Faster command responses or reduced resource use
- [ ] 🕵️ Logging/debugging – Improved diagnostics, logs, or debug output
- [ ] 🔧 Tooling – Scripts, benchmarks, or local dev improvements
- [ ] 🔒 Security fix – Permissions, validation, or Discord API-related vulnerabilities
- [ ] 🧰 Dependency update – Library or package updates

---

## 🧪 How Has This Been Tested?

- [ ] Unit Tests (`tests/unit/`)
- [ ] Integration Tests (`tests/integration/`)
- [x] Manual testing in Discord
- [ ] CI

> **Test Environment:**
> - OS: (e.g., `Linux`)
> - Python: (e.g., `3.12`)
> - Discord.py version: (e.g., `2.6.3`)
> - [ ] Docker build tested

---

## ✅ Checklist

- [x] My code follows repo [CONTRIBUTING.md](https://github.com/pesu-dev/discord_bot/blob/main/.github/CONTRIBUTING.md) guidelines
- [x] Self-review completed
- [ ] Added/updated comments and docstrings
- [ ] Updated relevant docs (README, help text, etc.)
- [x] No new warnings or errors introduced
- [ ] Added/updated tests
- [ ] All tests pass locally
- [ ] Ran linting and formatting
- [ ] Docker image builds and runs
- [ ] Changes are backwards compatible (if applicable)
- [ ] Feature flags or `.env` vars updated (if applicable)
- [ ] Tested in multiple environments (if applicable)

---

## 🛠️ Affected Bot Areas

- [x] Commands (e.g., `/ping`, `/help`)
- [ ] Event handling (on_message, on_ready, etc.)
- [ ] Permissions/roles
- [ ] Database/Storage
- [ ] Discord API/Integrations
- [ ] Bot startup/shutdown
- [ ] CI / Git hooks
- [ ] Dockerfile / build
- [ ] Dependencies (`pyproject.toml`)
- [ ] Tooling/scripts
- [ ] Other (specify in Additional Notes)

---

## 📸 Screenshots / Demos (if applicable)

---

## 🧠 Additional Notes

dotpmm committed:
Enhance 'purge' command with additional options

Updated the 'purge' command to allow deletion by amount, date, or message link.

dotpmm committed:
Refactor purge command for better readability

Fixed linting errors

sach-12 commented:
@dotpmm install githooks. Check step 3 [here](https://github.com/pesu-dev/discord_bot/blob/dev/.github/CONTRIBUTING.md#setting-up-your-environment)

dotpmm committed:
Refactor purge function for cleaner syntax

Fixed the typo and linting error

dotpmm committed:
Enhance 'purge' command with additional options

LINTING ERROR!!

dotpmm committed:
Fixed errors

dotpmm committed:
Hopefully fixed all errors

dotpmm commented:
@sach-12 Done?

sach-12 commented:
Why are there modifications in the `.githooks` files?

sach-12 commented on cogs/interactions/slash/mod.py:
```diff
@@ -540,16 +550,48 @@ async def purge(self, interaction: discord.Interaction, amount: int) -> None:
 
-        if amount < 1 or amount > 100:
-            await interaction.followup.send(content="Please specify a number between 1 and 100", ephemeral=True)
+        if sum(x is not None for x in [amount, message_link, date]) != 1:
```
Is it possible to use `any` here instead of `sum`?

dotpmm replied:
I used sum because we need to make sure exactly one of the three parameters is provided. any would only tell us if at least one is provided, which still allows two or three arguments at the same time(it just allows, we can find a workaround to select only one out of it, but then it defeats the purpose) . Using sum(x is not None …) lets us count how many were given and enforce the requirement cleanly.

dotpmm committed:
Reverted .githooks

dotpmm committed:
Add `/ban` command

Implemented the /ban slash command with proper permission checks,
DM notifications, logging to mod channel, and error handling.

dotpmm committed:
Revert "Add `/ban` command"

This reverts commit df897e1402ab12166059269d0af99ed9259eee96.
This was not meant to be in this PR
